### PR TITLE
fix(navigation): links work fine and are avaiable in the markup

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

Second attempt for making navigations links available for SEO.
Despite being present on the markup, they should work as expected when clicking on them.

Related with #4358

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
* **#4366** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4366?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @guilhermespopolin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/blocks/HeaderBlock.tsx [hunk 1]
```diff
@@ -89,18 +89,18 @@ export const NestedNavContainerBlock = ({ blok }: NestedNavContainerBlockProps)
           <NavigationMenuPrimitive.Sub defaultValue={blok.name}>
             <NavigationMenuPrimitive.List className={navigationSecondaryList}>
               {filteredNavItems.map((nestedBlock) => (
-                <SecondaryNavigationLink
+                <NavigationMenuPrimitive.Item
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/HeaderBlock.tsx:92]: This is not directly related. Just take to opportunity to fix the rendered structure. The way we had it before - with the  `SecondaryNavigationLink` rendering the `avigationMenuPrimitive.Item` would result on the following structured:

```
<ul>
  <a>
    <li>some text</li>
  </a>
</ul>
```

What we actually want is:

```
<ul>
  <li>
    <a>some text</a>
  </li>
</ul>
```

Comment (Youakeem) [reply apps/store/src/blocks/HeaderBlock.tsx:92]: You are awesome!

**Review by Youakeem — APPROVED** [review]

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jun 13, 10:05 AM EDT**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4366).
